### PR TITLE
Init plugin

### PR DESCRIPTION
**Pull request by TheBreaken**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        "phpunit/phpunit": "^5.7",
+        "shopware/shopware": "^5.7",
+        "solutiondrive/sd-test-environment-shopware": "~0.2",
+        "solutiondrive/standalone-build-tools": "^2.0"
+    },
+    "config": {
+        "sort-packages": true,
+        "standalone-build-tools": {
+            "phpspec-standalone.php7.4.phar": "http://build-tools.cloud.solutiondrive.de/phar/phpspec-standalone.php7.4.phar",
+            "ecs-standalone.phar": "http://build-tools.cloud.solutiondrive.de/phar/coding-standard-standalone.ecs.php7.3.phar",
```
Comment (sd-lueckel) [inline composer.json]:
```suggestion
            "ecs-standalone.phar": "http://build-tools.cloud.solutiondrive.de/phar/coding-standard-standalone.ecs.php7.4.phar",
```

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            "easy-coding-standard.yml": "http://build-tools.cloud.solutiondrive.de/phar/easy-coding-standard-php7.3.yml"
```
Comment (sd-lueckel) [inline composer.json]:
```suggestion
            "ecs.php": "http://build-tools.cloud.solutiondrive.de/phar/ecs7.4.php"
```

File: etc/easy-coding-standard.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+imports:
```
Comment (sd-lueckel) [inline etc/easy-coding-standard.yml]: Replace this file with:
etc/ecs.php
`<?php

declare(strict_types=1);

use Symplify\EasyCodingStandard\Config\ECSConfig;

return static function (ECSConfig $ecsConfig): void {
    $ecsConfig->import(__DIR__ . '/../vendor/bin/ecs.php');
};
`

File: etc/scripts/checkCodingStandards.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+if [ "0" != "$(echo $?)" ]; then
+    echo "No php files found to lint. So technically spoken everything's fine, but please check that I haven't missed any files."
+    exit 0
+fi
+
+SEPERATOR=" "
+FOLDERS=$(printf "${SEPERATOR}%s" "${FILTERED_FOLDERS[@]}")
+
+vendor/bin/ecs-standalone.phar check --no-progress-bar -n -c etc/easy-coding-standard.yml $FOLDERS $@
```
Comment (sd-lueckel) [inline etc/scripts/checkCodingStandards.sh]:
```suggestion
vendor/bin/ecs-standalone.phar check --no-progress-bar -n -c etc/ecs.php $FOLDERS $@
```

File: plugin.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <label lang="de">nlxWebpOptimizer</label>
+    <label lang="en">nlxWebpOptimizer</label>
+    <version>1.0.0</version>
+    <copyright>(c) by netlogix GmbH &amp; Co. KG</copyright>
+    <license>Proprietary</license>
+    <link>https://websolutions.netlogix.de/</link>
+    <author>netlogix GmbH &amp; Co. KG</author>
+
+    <description lang="de">
+        Plugin, um automatisch Webp Varianten der Media-Assets zu generieren
```
Comment (sd-lueckel) [inline plugin.xml]:
```suggestion
        Plugin, um automatisch WebP Varianten der Media-Assets zu generieren
```

File: plugin.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    </description>
+    <description lang="en">
+        Plugin to automatically generate webp variants of the media assets
```
Comment (sd-lueckel) [inline plugin.xml:16]:
```suggestion
        Plugin to automatically generate WebP variants of the media assets
```

File: composer.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
+    "name": "netlogix/nlxwebpoptimizer",
+    "description": "Plugin to automatically generate webp variants of the media assets",
```
Comment (sd-lueckel) [inline composer.json:3]:
```suggestion
    "description": "Plugin to automatically generate WebP variants of the media assets",
```

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    "type": "shopware-plugin",
+    "license": "proprietary",
+    "authors": [
+        {
+            "name": "netlogix GmbH & Co. KG",
+            "email": "websolutions@netlogix.de"
+        }
+    ],
+    "extra": {
+        "installer-name": "nlxWebpOptimizer"
```
Comment (sd-lueckel) [inline composer.json]:
```suggestion
        "installer-name": "nlxWebPOptimizer"
```

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    "autoload": {
+        "psr-4": {
+            "nlxWebpOptimizer\\": "."
```
Comment (sd-lueckel) [inline composer.json]:
```suggestion
            "nlxWebPOptimizer\\": "."
```

File: nlxWebpOptimizer.php [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<?php
```
Comment (sd-lueckel) [inline nlxWebpOptimizer.php]: WebP is a name and should be written as WebP

File: phpspec.yml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+suites:
+  default:
+    namespace: nlxWebpOptimizer
```
Comment (sd-lueckel) [inline phpspec.yml]:
```suggestion
    namespace: nlxWebPOptimizer
```

File: phpspec.yml [hunk 12]
```diff
@@ -0,0 +1,6 @@
...
+    psr4_prefix: nlxWebpOptimizer
```
Comment (sd-lueckel) [inline phpspec.yml]:
```suggestion
    psr4_prefix: nlxWebPOptimizer
```

File: phpunit.xml.dist [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+         colors="true"
+         backupGlobals="true"
+         beStrictAboutOutputDuringTests="true"
+         beStrictAboutResourceUsageDuringSmallTests="true"
+         beStrictAboutTestsThatDoNotTestAnything="true"
+         beStrictAboutTodoAnnotatedTests="true"
+         verbose="true">
+
+    <testsuites>
+        <testsuite name="nlxWebpOptimizer_AllTests">
```
Comment (sd-lueckel) [inline phpunit.xml.dist]:
```suggestion
        <testsuite name="nlxWebPOptimizer_AllTests">
```

File: plugin.xml [hunk 14]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<plugin xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../engine/Shopware/Components/Plugin/schema/plugin.xsd">
+    <label lang="de">nlxWebpOptimizer</label>
```
Comment (sd-lueckel) [inline plugin.xml]:
```suggestion
    <label lang="de">nlxWebPOptimizer</label>
```

File: plugin.xml [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    <label lang="en">nlxWebpOptimizer</label>
```
Comment (sd-lueckel) [inline plugin.xml]:
```suggestion
    <label lang="en">nlxWebPOptimizer</label>
```

File: tests/Functional/PluginTest.php [hunk 16]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace nlxWebpOptimizer\Tests;
```
Comment (sd-lueckel) [inline tests/Functional/PluginTest.php]:
```suggestion
namespace nlxWebPOptimizer\Tests;
```

File: tests/Functional/PluginTest.php [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+use nlxWebpOptimizer\nlxWebpOptimizer as Plugin;
```
Comment (sd-lueckel) [inline tests/Functional/PluginTest.php]:
```suggestion
use nlxWebPOptimizer\nlxWebPOptimizer as Plugin;
```

File: tests/Functional/PluginTest.php [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+use Shopware\Components\Test\Plugin\TestCase;
+
+class PluginTest extends TestCase
+{
+    protected static $ensureLoadedPlugins = [
+        'nlxWebpOptimizer' => []
```
Comment (sd-lueckel) [inline tests/Functional/PluginTest.php]:
```suggestion
        'nlxWebPOptimizer' => []
```

File: tests/Functional/PluginTest.php [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+    ];
+
+    public function testCanCreateInstance()
+    {
+        /** @var Plugin $plugin */
+        $plugin = Shopware()->Container()->get('kernel')->getPlugins()['nlxWebpOptimizer'];
```
Comment (sd-lueckel) [inline tests/Functional/PluginTest.php]:
```suggestion
        $plugin = Shopware()->Container()->get('kernel')->getPlugins()['nlxWebPOptimizer'];
```

**Review by sd-lueckel — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (TheBreaken) [reply composer.json]: Done

File: composer.json (hunk 2 above)
Comment (TheBreaken) [reply composer.json]: Done

File: etc/easy-coding-standard.yml (hunk 3 above)
Comment (TheBreaken) [reply etc/easy-coding-standard.yml]: Done.

File: etc/scripts/checkCodingStandards.sh (hunk 4 above)
Comment (TheBreaken) [reply etc/scripts/checkCodingStandards.sh]: Done.

File: plugin.xml (hunk 5 above)
Comment (TheBreaken) [reply plugin.xml]: Done.

File: plugin.xml (hunk 15 above)
Comment (TheBreaken) [reply plugin.xml]: Done.

File: tests/Functional/PluginTest.php (hunk 16 above)
Comment (TheBreaken) [reply tests/Functional/PluginTest.php]: Done.

File: tests/Functional/PluginTest.php (hunk 17 above)
Comment (TheBreaken) [reply tests/Functional/PluginTest.php]: Done.

File: tests/Functional/PluginTest.php (hunk 18 above)
Comment (TheBreaken) [reply tests/Functional/PluginTest.php]: Done.

File: tests/Functional/PluginTest.php (hunk 19 above)
Comment (TheBreaken) [reply tests/Functional/PluginTest.php]: Done.

**Review by sd-lueckel — APPROVED** [review]

**Pull request merged**